aten/src/ATen/native/TensorShape.cpp

```diff
@@ -413,6 +413,11 @@ Tensor& set_storage_meta__symint(Tensor& result, Storage storage, c10::SymInt st
 
   // Matches maybe_resize_storage_cpu no-numel behavior
   if (TORCH_GUARD_SIZE_OBLIVIOUS(result.sym_numel().sym_ne(0))) {
+    // Following checks are implied by the condition above but
+    // cannot be inferred due to incompleteness of symbolic reasoning
+    for (auto i = 0; i < result.dim(); ++i) {
+      TORCH_SYM_CHECK(result.sym_size(i).sym_ne(0), "Size must be non-zero");
```

Comment:
I don't think this is sound. When TORCH_GUARD_SIZE_OBLIVIOUS is used, this means that at runtime, the size may potentially be zero (and we are asserting that the code on the inside of this conditional is valid even when sizes are zero). But if you turn around and immediately assert that the sizes are not zero (no size oblivious, so you'll end up with runtime asserts), then you've violated that contract.

Reply:
Undone.